docs/grammar.md

```diff
@@ -59,26 +59,37 @@ Terminals are used to match text into symbols. They can be defined as a combinat
 **Syntax:**
 
 ```html
-<NAME> [. <priority>] : <literals-and-or-terminals>
+<NAME> [. <priority>] : <items-to-match>
 ```
 
-Terminal names must be uppercase.
+Terminal names must be uppercase.  They must start with an underscore (`_`) or a letter (`A` through `Z`), and may be composed of letters, underscores, and digits (`0` through `9`).  Terminal names that start with "_" will not be included in the parse tree, unless the `keep_all_tokens` option is specified.
 
 Literals can be one of:
 
-* `"string"`
-* `/regular expression+/`
-* `"case-insensitive string"i`
-* `/re with flags/imulx`
-* Literal range: `"a".."z"`, `"1".."9"`, etc.
+* Literal range: `"a".."z"`, `"1".."9"`, etc. - Each literal must be a single character, and the range represends all values between the two literals, inclusively.
 
-Terminals also support grammar operators, such as `|`, `+`, `*` and `?`.
+Each item is one of:
```

Comment:
This list already appears below. Why duplicate it?

Reply:
The item list for terminals is different from the list later for rules, in that a terminal's items cannot include rules.  But yeah, it's easy to overlook that in the big wall of text.

Honestly, if I was writing this document from scratch, I'd have written the full grammar in BNF and avoided the simplifying statements.  But I assume your audience here is people without a CS background who may never have written a parser by hand, and who might find BNF opaque.

Reply:
I'm not following the reasoning. Why not have one list, and in rules write: "an item can be everything in that list, plus a rule". Or even put rule as part of the list, and mention that it can't appear in terminals.

Reply:
Fixed by 83a374f5d00cd1f6bf3b43d5bf799cf7ef804cb3.